shanoir-ng-datasets/src/main/java/org/shanoir/ng/solr/controler/SolrApiController.java

```diff
@@ -47,10 +51,21 @@ public class SolrApiController implements SolrApi {
 	
 	@Autowired
 	private SolrService solrService;
-	
+
+	@Autowired
+	private ShanoirEventService eventService;
+
 	@Override
-	public ResponseEntity<Void> indexAll() throws RestServiceException, SolrServerException, IOException {
-		solrService.indexAll();		
+	public ResponseEntity<Void> indexAll() throws SolrServerException, IOException {
+		ShanoirEvent event = new ShanoirEvent(
```

Comment:
Can't we publish this event directly in indexAll() method to avoid adding a new  ShanoirEventService dependency? And then merge both indexAll() and indexAll(event) methods ?
(and for the scheduled method, we may also publish events ? I don't know if it's feasible, tell me)